src/MessageProducer.php

```diff
@@ -1,26 +1,25 @@
 <?php
 /*
- * This file is part of the codeliner/psb-bernard-dispatcher.
- * (c) Alexander Miertsch <kontakt@codeliner.ws>
+ * This file is part of the prooph/psb-bernard-dispatcher.
+ * (c) 2014 - 2015 prooph software GmbH <contact@prooph.de>
  *
  * For the full copyright and license information, please view the LICENSE
  * file that was distributed with this source code.
- * 
- * Date: 31.10.14 - 15:08
+ *
+ * Date: 10/31/14 - 03:08 PM
  */
 namespace Prooph\ServiceBus\Message\Bernard;
 
 use Bernard\Producer;
-use Prooph\Common\Messaging\RemoteMessage;
-use Prooph\ServiceBus\Message\RemoteMessageDispatcher;
+use Prooph\Common\Messaging\Message;
 
 /**
- * Class MessageDispatcher
+ * Class MessageProducer
  *
  * @package Prooph\ServiceBus\Message\Bernard
  * @author Alexander Miertsch <kontakt@codeliner.ws>
  */
-class MessageDispatcher implements RemoteMessageDispatcher
+class MessageProducer
```

Comment:
Are you sure about removing the interface completely? The interface can also be something like:

    public function __invoke(Message $message);

If you don't want to make the interface a hard dependency, we can do something simliar to what the ZF2's module manager does.

    if (!$producer instanceof Producer && !is_callable($producer)) {
        throw new Exception\InvalidArgumentException();
    }

see: https://github.com/zendframework/zend-modulemanager/blob/master/src/Listener/ConfigListener.php#L120